### ISSUE TITLE
ResidualsPlot: "UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names"

### ISSUE THREAD

**Issue by luukburger** [DistrictDataLabs/yellowbrick#1261]
**Describe the bug**
I'm running a ResidualsPlot on an LinearRegression Estimator that is already fitted using sklearn. The Estimator was fitted using an X_train DataFrame including column names, and a y_train Series which is also named.

Now when I draw a ResidualsPlot on the fitted model using yellowbrick, I get an sklearn warning in my notebook:

`~/opt/miniconda3/envs/py39ds/lib/python3.9/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names`

The warning is not thrown when I fit using yellowbrick, but I'd rather use yellowbrick only for drawing the ResidualsPlot and not change the rest of my notebook.

**To Reproduce**
Warning thrown with this code:
```python
lin_reg = LinearRegression()
lin_reg.fit(X_train_scaled, y_train)

fig, ax = plt.subplots(figsize=(15, 10))
_ = residuals_plot(estimator=lin_reg,
                   is_fitted=True,
                   ax=ax,
                   X_train=X_train_scaled,
                   y_train=y_train,
                   X_test=X_test_scaled,
                   y_test=y_test)
```

Warning not thrown with this code:
```python
fig, ax = plt.subplots(figsize=(15, 10))
_ = residuals_plot(LinearRegression(), ax=ax, X_train=X_train_scaled, y_train=y_train, X_test=X_test_scaled, y_test=y_test)
```

**Expected behavior**
The warning says "X does not have valid feature names" but it does, so I don't understand why it's thrown.

**Desktop**
 - OS: macOS 12.4
 - IDE: VS Code 1.68.1 + Jupyter Extension v2022.5.1001601848 + miniconda 4.13
 - Python 3.9.12 + yellowbrick 1.4

**Comment by lwgray** [comment]
@luukburger I tried to replicate your issue but had no luck.  Hower, I think I know what the problem is.  Here a link to a StackOverflow explanation https://stackoverflow.com/questions/69326639/sklearn-warning-valid-feature-names-in-version-1-0

Thanks for using Yellowbrick

**Comment by luukburger** [comment]
@lwgray

Strange that you cannot reproduce. I just pasted the standalone code in a deepnote.com iPython notebook and there also the warning is thrown.

1. Sorry, in my initial post I copied the code directly from my notebook. In my second post I quickly created some standalone code and used different variable names. The "_scaled" variables in the initial post are DataFrame's too.
2. You're right, the warning is thrown by sklearn itself, but that gets called by yellowbrick, right? In Method 1 and Method 2 I try to make yellowbrick use a model that is already fitted using sklearn, and then yellowbrick seems to require that the fitting was done with `numpy.ndarray` input. Maybe that's just what it is, but since sklearn allows fitting with `pandas.DataFrame`, I thought it's maybe something in the way yellowbrick calls sklearn.

**Comment by lwgray** [comment]
What version of scikit-learn are you using?

**Comment by luukburger** [comment]
Package versions on my local machine:

```
ipython                   8.3.0
matplotlib                3.5.1
pandas                    1.4.2
python                    3.9.12
scikit-learn              1.0.2
yellowbrick               1.4
```

**Comment by luukburger** [comment]
In my standalone code-snippet from the comment above I added a fifth method to demonstrate the problem. If I train using scikit with DataFrames/Series, and then explicitly call residuals plot from yellowbrick using ndarrays, then I would indeed expect the warning. But I would expect it 4 times (one for each of the arguments of residuals plot: X_train, y_train, X_test, y_test).

I get it 5 times though, so I think we're looking for the cause of the fifth one (which is the same one as the single one form method 1).

**Comment by lwgray** [comment]
Hi @luukburger .  This issue is fixed in our development branch.  You can pip install that branch and the issue should be resolved.

**Comment by luukburger** [comment]
Oh that's great, thanks a lot for helping out.

I'm quite new to Python and ML, so I'm not sure how to update to a development branch. I'm also using Conda instead of pip. I'll update when a new version is released.

**Comment by bbengfort** [comment]
@lwgray can we close this issue?

**Comment by bbengfort** [comment]
@luukburger we just released Yellowbrick v1.5 -- please update to the latest version and this fix should be addressed!

**Comment by luukburger** [comment]
I just updated to v1.5 and the issue is indeed fixed. Thanks a lot!
